### ISSUE TITLE
[yggtorrent] Exception (yggtorrent): Login failed: No form found on http://www2.yggtorrent.se/ using form selector 

### ISSUE THREAD

**Issue by cadavor** [Jackett/Jackett#7505]
### Environment

**OS**: Freenas 13-U1 (FreeBSD 11) in Jail

**.Net Runtime**: Mono 

**.Net Version**: 5.10.1.57

**Jackett Version**: 0.13.446.0

**Last Working Jackett Version**: Yes but don't know which one

**Are you using a proxy or VPN?** [yes/no] yes VPN

### Description

Access throw same VPN server on Chrome works well.
I found the #user-login element by inspecting sourcecode of website

### Logged Error Messages

```
Jackett.Common.IndexerException: Exception (yggtorrent): Login failed: No form found on http://www2.yggtorrent.se/ using form selector #user-login ---> System.Exception: Parse error ---> Jackett.Common.ExceptionWithConfigData: Login failed: No form found on http://www2.yggtorrent.se/ using form selector #user-login
  at Jackett.Common.Indexers.CardigannIndexer+<DoLogin>d__16.MoveNext () [0x004b6] in <96056e77de624e99b100b6df925866d5>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Jackett.Common.Indexers.CardigannIndexer+<PerformQuery>d__29.MoveNext () [0x009ce] in <96056e77de624e99b100b6df925866d5>:0 
   --- End of inner exception stack trace ---
  at Jackett.Common.Indexers.BaseWebIndexer.OnParseError (System.String results, System.Exception ex) [0x00064] in <96056e77de624e99b100b6df925866d5>:0 
  at Jackett.Common.Indexers.CardigannIndexer+<PerformQuery>d__29.MoveNext () [0x01c05] in <96056e77de624e99b100b6df925866d5>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Jackett.Common.Indexers.BaseIndexer+<ResultsForQuery>d__69.MoveNext () [0x00091] in <96056e77de624e99b100b6df925866d5>:0 
   --- End of inner exception stack trace ---
  at Jackett.Common.Indexers.BaseIndexer+<ResultsForQuery>d__69.MoveNext () [0x000ee] in <96056e77de624e99b100b6df925866d5>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Jackett.Common.Indexers.BaseWebIndexer+<ResultsForQuery>d__25.MoveNext () [0x0006b] in <96056e77de624e99b100b6df925866d5>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Jackett.Common.Services.IndexerManagerService+<TestIndexer>d__18.MoveNext () [0x000a3] in <96056e77de624e99b100b6df925866d5>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Jackett.Server.Controllers.IndexerApiController+<Test>d__15.MoveNext () [0x000e1] in <38c45ba08bf94c03a70200e9d95fbc20>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor+TaskOfIActionResultExecutor+<Execute>d__0.MoveNext () [0x00071] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+<InvokeActionMethodAsync>d__12.MoveNext () [0x00131] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+<InvokeNextActionFilterAsync>d__10.MoveNext () [0x0009e] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow (Microsoft.AspNetCore.Mvc.Filters.ActionExecutedContext context) [0x0001b] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next (Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+State& next, Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+Scope& scope, System.Object& state, System.Boolean& isCompleted) [0x00382] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+<InvokeInnerFilterAsync>d__13.MoveNext () [0x0002f] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+<InvokeNextResourceFilter>d__23.MoveNext () [0x0009f] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow (Microsoft.AspNetCore.Mvc.Filters.ResourceExecutedContext context) [0x0001b] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next (Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+State& next, Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+Scope& scope, System.Object& state, System.Boolean& isCompleted) [0x00840] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+<InvokeFilterPipelineAsync>d__18.MoveNext () [0x0002f] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+<InvokeAsync>d__16.MoveNext () [0x0012e] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Builder.RouterMiddleware+<Invoke>d__4.MoveNext () [0x001cb] in <6092a16d93814eba828b517a2b132f80>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware+<Invoke>d__6.MoveNext () [0x00384] in <427697fe42b7459ba5302fb76d339d3b>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware+<Invoke>d__7.MoveNext () [0x0043e] in <f352e566abf6421e87eafbcf57a0b237>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <0e06830de9a44394a7e366951eabca52>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <0e06830de9a44394a7e366951eabca52>:0 
  at Jackett.Server.Middleware.CustomExceptionHandler+<Invoke>d__3.MoveNext () [0x0008a] in <38c45ba08bf94c03a70200e9d95fbc20>:0 
```

### Screenshots

No

**Comment by garfield69** [comment]
Due  to the way the site handles the login we cannot reliably trap all the login errors it generates :-(

Usually this type of error means that your login was rejected by yggtorrent.

So use your browser to login to the site to confirm your account credentials, and then check that you have a good ratio, and that you do not have any outstanding alerts, notices or unread mail.
Please report what you find.

**Comment by cadavor** [comment]
As I mentionned, no problem to login with my browser.
Account valid and active (good ratio), no alert, no notices and no unread email.

**Comment by garfield69** [comment]
in that case please provide the full enhanced log. this will include the html and other messages prior to the final error, that may provide  a clue to what is going on at your end.

Please provide the **full enhanced log** file. 
1. scroll down to the bottom of the Jackett Dashboard and *tick* the *enhanced Logging* checkbox
2. scroll up a bit and *click* on the *apply server settings* button
3. repeat the test
4. find the *log.txt* file (linux `~/.config/Jackett/`, windows `%ProgramData%\Jackett`)
5. edit it with a plain text editor, redacting any personal details, usernames, passwords, passkeys, hashes etc.
6. save, then drag-drop it here for us to take a look at.

(Note: When you restart Jackett the enhanced-log switch will be turned off. This prevents performance hits during normal operations).

Thanks.

**Comment by garfield69** [comment]
actually, never mind, the clue is in the header ;-)

please read and apply https://github.com/Jackett/Jackett/issues/7095#issuecomment-582238746
I expect your existing sitelink and/or search links are not quite correct.
let me know how you go.

**Comment by cadavor** [comment]
I've read and apply https://github.com/Jackett/Jackett/issues/7095#issuecomment-582238746
and it's fine now.

Thanks for help

**Comment by cadavor** [comment]
In fact, I noticed that sometimes the test failed (login problem or account deactivation due to the ratio) and that Jackett automatically updated the address with the redirection url captured by the error (www2.yggtorrent.se). So, on the next test, when the problem is solved, the test fails because of the wrong address replaced.
Maybe, it will be better to ask if the address should be replaced in case of error no?

**Comment by garfield69** [comment]
Rant begins:
yggtorrent is a pain in the ass. it is the only site that uses two domains at one time, one to login and another to search. So we tried to make allowances within the scope of the standard methods that all the other sites conform to.

this is why we currently have the followredirect to allow the yggtorrent indexer to work. But it comes at a price, due to the way the site abuses the redirection process.

The site's login is also rubbish. it is set up to provide error messages for certain failures, which we trap, but then it uses none of them, and just provides a redirect to another page entirely, which we cannot trap.

The only time it would probably be appropriate to do a popup asking permission to alter the sitelink is when the user is using the Jackett dashboard.
But this is actually a small number of usage cases. The majority of users access Jackett via other apps like Sonarr, Radarr, DuckieTV etc. and its not appropriate to request permission with a popup in these cases.

Basically what I am saying is: this site is the only one that behaves in this way, and it is not worth our time to try to mediate the login-failed-because-of-account-issues-pop-a-question-to-prevent-altering-redirected-site-link for just this one poorly designed website.

Frankly, if it wasn't so bloody popular with the French users, I'd delete the darn thing and be free of the headaches it frequently causes, especially when they change their domains so damn regularly.

Rant over. Forgive me for venting.
